environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentStopService.java

```diff
@@ -57,4 +66,16 @@ private void validateStoppable(EnvironmentDto environment, String accountId) {
             throw new BadRequestException(message);
         }
     }
+
+    private void validaFreeIpaIsNotRunningOnSpotInstances(EnvironmentDto environment) {
+        Integer freeIpaSpotPercentage = Optional.ofNullable(environment.getFreeIpaCreation())
+                .map(FreeIpaCreationDto::getAws)
+                .map(FreeIpaCreationAwsParametersDto::getSpot)
+                .map(FreeIpaCreationAwsSpotParametersDto::getPercentage)
+                .orElse(0);
+
+        if (freeIpaSpotPercentage != 0) {
+            throw new BadRequestException("The environment can not be stopped because FreeIpa is running on spot instances.");
```

Comment:
probably the environment name should be fine in the exception message.

Reply:
Added